javascript2-week2/quizApp2.js

```diff
@@ -0,0 +1,184 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const quizForm = document.getElementById('quizForm');
+    const questionsList = document.getElementById('questionsList');
+    const filter = document.getElementById('filter');
+    let quizQuestions = [];
+
+    quizForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        if (validateForm()) {
+            addQuestion();
+        }
+    });
+
+    document.getElementById('randomize').addEventListener('click', randomizeOptions);
+    document.getElementById('correctAnswer').addEventListener('change', updateOptionColors);
+
+    filter.addEventListener('input', filterQuestions);
+
+    function validateForm() {
+        const question = document.getElementById('question').value;
+        const options = Array.from(document.getElementsByClassName('option'));
+        const hasEmptyOption = options.some(opt => opt.value.trim() === '');
+
+        if (!question.trim() || hasEmptyOption) {
+            alert('Please fill in all fields.');
+            return false;
+        }
+
+        if (question.length > 140) {
+            alert('Question length must not exceed 140 characters.');
+            return false;
+        }
+
+        return true;
+    }
+
+    function addQuestion() {
+        const question = document.getElementById('question').value;
```

Comment:
You are accessing those elements mutliple times - maybe you could have a funcion: `getQuestion` that will access the elements and return the quesion object?